### PR TITLE
refactor(web): VisualKeyboard - key event model

### PR DESCRIPTION
**Pull request by jahorton**
One of the goals for #5023 involves having the OSK generate events for keystrokes that are then handled outside of the OSK.  While we're not quite there yet...

- This PR isolates the (`PreProcessor`) code that would be in a first-pass event handler for the OSK's key events, moving the code that should be OSK-internal into the `VisualKeyboard` class.
  - All code remaining within `com.keyman.osk.PreProcessor` will be outside of the OSK-Core.
- The only 'external' dependency of the newly-internalized code: Web-Core classes.
  - The sole use for `com.keyman.singleton` in the affected codepaths is to access `keyman.core`, which is not _currently_ available via class method or field within `VisualKeyboard`.  That'll be resolved further down the line.

### CODE REVIEW

File: web/source/osk/preProcessor.ts [hunk 1]
```diff
@@ -30,50 +9,39 @@ namespace com.keyman.osk {
...
       let keyman = com.keyman.singleton;
       var Lelem = keyman.domManager.getLastActiveElement();
 
       if(Lelem != null) {
         // Handle any DOM state management related to click inputs.
         let outputTarget = dom.Utils.getOutputTarget(Lelem);
         keyman.domManager.initActiveElement(Lelem);
-  
-        // Turn off key highlighting (or preview)
-        keyman['osk'].vkbd.highlightKey(e,false);
```
Comment (mcdurdin) [inline web/source/osk/preProcessor.ts:43]: Why is this no longer needed?

**Review by mcdurdin — APPROVED** [review]
I think this LGTM, barring that one question.

File: web/source/osk/preProcessor.ts (hunk 1 above)
Comment (jahorton) [reply web/source/osk/preProcessor.ts:43]: Elsewhere in this PR: https://github.com/keymanapp/keyman/blob/28fe887f84c2936708f846c7ae88fb5d856ba879/web/source/osk/visualKeyboard.ts#L1543-L1545

`initKeyEvent` always occurs for any OSK-based key event and will run first.  As a result, this deletion removes a redundancy.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.69-alpha](https://keyman.com/downloads/releases/alpha/15.0.69)

**Pull request merged**